streams/src/main/java/org/apache/kafka/streams/Topology.java

```diff
@@ -519,30 +519,32 @@ public synchronized Topology addSink(final String name,
 
     /**
      * Add a new sink that forwards records from upstream parent processor and/or source nodes to Kafka topics dynamically.
+     * The topics that it may ever send to should be pre-created.
      * The sink will use the {@link StreamsConfig#DEFAULT_KEY_SERDE_CLASS_CONFIG default key serializer} and
      * {@link StreamsConfig#DEFAULT_VALUE_SERDE_CLASS_CONFIG default value serializer} specified in the
      * {@link StreamsConfig stream configuration}.
      *
      * @param name              the unique name of the sink
-     * @param topicExtractor    the mapper to dynamically choose the name of the Kafka topic to which this sink should write per reach record
+     * @param topicExtractor    the extractor to determine the name of the Kafka topic to which this sink should write for reach record
      * @param parentNames       the name of one or more source or processor nodes whose output records this sink should consume
-     *                          and write to its topic
+     *                          and dynamically write to topics
      * @return                  itself
-     * @throws TopologyException if parent processor is not added yet, or if this processor's name is equal to the parent's name
+     * @throws TopologyException if parent processor is not added yet, or if this processor's name is not unique
```

Comment:
I think we throw for all three cases.

Reply:
`this processor's name is equal to the parent's name` is subsumed by `this processor's name is not unique`, right?

Reply:
I don't think so. If `A` connects to itself, it's still a single node:
```
topology.addSource("source",...);
topology.addProcessor("name", ..., "parent"); // throws because "parent" no added yet
topology.addProcessor("source", ...); // throws because "source" is already used
topology.addProcessor("processor", ..., "processor"); // throws because it connects to itself --- note that internally, we would add "processor" first before setting up the connection -- we do have an additional check in the code to throw of people build a loop like this
```

Reply:
Make sense, will add it back.